Latest cel-go breaks tests in pkg/cel

### Description

**Observed Behavior**:
After updating `github.com/google/cel-go` to latest, there are failing tests.

**Expected Behavior**:
We should stay abreast with upstream CEL changes.

This currently blocks e.g. upgrading `sigs.k8s.io/controller-runtime` which has a dependency on `cel-go`.

**Reproduction Steps**
```sh
go get -u github.com/google/cel-go
go mod tidy
go test ./pkg/...
```

**Versions**:
- kro version: `main`

**Error Logs** (if applicable)**:
```
λ go test ./pkg/... -count=1
?       github.com/kro-run/kro/pkg      [no test files]
?       github.com/kro-run/kro/pkg/cel  [no test files]
--- FAIL: TestInspector_UnknownResourcesAndCalls (0.00s)
    --- FAIL: TestInspector_UnknownResourcesAndCalls/filter_with_multiple_conditions (0.00s)
        inspector_test.go:511: UnknownResources = [{@result @result} {@result @result} {@result @result}], want []
    --- FAIL: TestInspector_UnknownResourcesAndCalls/ambiguous_i_usage_-_both_resource_and_loop_var (0.00s)
        inspector_test.go:511: UnknownResources = [{@result @result} {@result @result} {@result @result}], want []
FAIL
FAIL    github.com/kro-run/kro/pkg/cel/ast      0.018s
?       github.com/kro-run/kro/pkg/client       [no test files]
?       github.com/kro-run/kro/pkg/controller/instance  [no test files]
ok      github.com/kro-run/kro/pkg/controller/instance/delta    0.006s
?       github.com/kro-run/kro/pkg/controller/resourcegraphdefinition   [no test files]
ok      github.com/kro-run/kro/pkg/dynamiccontroller    1.129s
ok      github.com/kro-run/kro/pkg/graph        0.060s
?       github.com/kro-run/kro/pkg/graph/crd    [no test files]
ok      github.com/kro-run/kro/pkg/graph/dag    0.006s
ok      github.com/kro-run/kro/pkg/graph/emulator       0.010s
ok      github.com/kro-run/kro/pkg/graph/fieldpath      0.003s
ok      github.com/kro-run/kro/pkg/graph/parser 0.006s
?       github.com/kro-run/kro/pkg/graph/schema [no test files]
?       github.com/kro-run/kro/pkg/graph/variable       [no test files]
ok      github.com/kro-run/kro/pkg/metadata     0.014s
?       github.com/kro-run/kro/pkg/requeue      [no test files]
ok      github.com/kro-run/kro/pkg/runtime      0.021s
ok      github.com/kro-run/kro/pkg/runtime/resolver     0.003s
ok      github.com/kro-run/kro/pkg/simpleschema 0.003s
?       github.com/kro-run/kro/pkg/testutil/generator   [no test files]
?       github.com/kro-run/kro/pkg/testutil/k8s [no test files]
FAIL
```

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

--- a/go.mod
+++ b/go.mod
@@ -5,7 +5,7 @@ go 1.24.0
 require (
 	github.com/go-logr/logr v1.4.2
 	github.com/gobuffalo/flect v1.0.2
-	github.com/google/cel-go v0.22.0
+	github.com/google/cel-go v0.24.1
 	github.com/onsi/ginkgo/v2 v2.20.0
 	github.com/onsi/gomega v1.34.1
 	github.com/prometheus/client_golang v1.19.1
@@ -25,7 +25,7 @@ require (
 )
 
 require (
-	cel.dev/expr v0.18.0 // indirect
+	cel.dev/expr v0.19.1 // indirect
 	github.com/antlr4-go/antlr/v4 v4.13.0 // indirect
 	github.com/beorn7/perks v1.0.1 // indirect
 	github.com/cespare/xxhash/v2 v2.3.0 // indirect

--- a/pkg/cel/ast/inspector.go
+++ b/pkg/cel/ast/inspector.go
@@ -493,7 +493,7 @@ func (a *Inspector) structExpressionToString(s *exprpb.Expr_CreateStruct) string
 }
 
 func isInternalIdentifier(name string) bool {
-	return name == "__result__" || strings.HasPrefix(name, "$$")
+	return name == "@result" || strings.HasPrefix(name, "$$")
 }
 
 func isInternalFunction(name string) bool {

